astropy/coordinates/earth.py

```diff
@@ -646,12 +651,45 @@ def get_itrs(self, obstime=None):
 
         # do this here to prevent a series of complicated circular imports
         from .builtin_frames import ITRS
-        return ITRS(x=self.x, y=self.y, z=self.z, obstime=obstime)
+        itrs_coo = ITRS(x=self.x, y=self.y, z=self.z, obstime=obstime)
+        if include_velocity:
+            zeros = np.broadcast_to(0. * u.km / u.s, (3,) + itrs_coo.shape, subok=True)
```

Comment:
It's the broadcasting that's slow here - not the creation of the `Quantity`.

```python
In [30]: %timeit np.broadcast_to(ZERO_SPEED, itrs.shape, subok=True)                                                                                        
79.3 µs ± 3.84 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [31]: %timeit np.broadcast_to(0. * (u.km / u.s), (3,) + itrs.shape, subok=True)                                                                          
72.2 µs ± 5.58 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

I've included the brackets here though

Reply:
Surprising!  I played around a bit and fastest is
```
np.broadcast_to(np.zeros(()), (3,)+itrs.shape) << u.km/u.s
```
In the end, though, it would all be negligible compared to the rest of the routine, so let's keep it as you have it now.

Reply:
I disagree. If there are easy gains **that don't harm legibility** we should always chose the most efficient solution. So replacing with your proposal is the way to go I think.

Reply:
Wait, is this just `np.zeros((3, ) + itrs_coo.shape) << u.km / u.s`? Why the broadcast for a single number?

Reply:
Save memory...

Reply:
Ha! TIL

Reply:
I find @mhvk's fastest solution not very legible, so I'll stick with it as it is now.